api/include/opentelemetry/nostd/span.h

```diff
@@ -240,9 +240,6 @@ class span<T, dynamic_extent>
   T *end() const noexcept { return data_ + extent_; }
 
 private:
-  // Note: matches libstdc++'s layout for std::span
```

Comment:
But it seems the definition here does provide the ABI compatibility as libc++, do you mean this is unnecessary?

Reply:
It was compatible with one version of libc++ at one point, but the library implementation changed other time.

We can't guarantee ABI compatibility with a moving target, so we should not claim or hint is is compatible.